### PR TITLE
Intercom

### PR DESCRIPTION
**Pull request by dooma**
Implemented(overstatement) Intercom.io.
screenshot from 2013-07-01 22 29 39

### CODE REVIEW

File: app/controllers/coursewareable/homes_controller.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def dashboard
       redirect_to root_path unless logged_in?
       timeline = current_user.activities_as_owner.reverse
       @timeline = Kaminari.paginate_array(timeline).page(params[:page])
+      intercom_custom_data.user[:app_activated_at] = Time.now
```
Comment (stas) [inline app/controllers/coursewareable/homes_controller.rb]: Nu e necesar, nu cred că ne ajută la ceva că știm când și-a activat intercom-ul. :)

Plus, că nu aș folosi intercom-ul doar decât în producție, și asta ar pica ceva teste.

File: config/initializers/intercom.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  # == Intercom API Key
+  # This is required for some Intercom rake tasks like importing your users;
+  # you can generate one at https://www.intercom.io/apps/api_keys.
+  #
+  config.api_key = "9ba7c361e3bcec79298b31c9ad5efae8942132d4"
+
+  # == Enabled Environments
+  # Which enviornments is auto inclusion of the Javascript enabled for
+  #
+  config.enabled_environments = ["development", "production"]
```
Comment (stas) [inline config/initializers/intercom.rb]: Just production is needed. In fact, you can comment out all this line.

Comment (stas) [inline config/initializers/intercom.rb]: Just production is needed. In fact, you can comment out all this line.

File: config/initializers/intercom.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  # == User Custom Data
+  # A hash of additional data you wish to send about your users.
+  # You can provide either a method name which will be sent to the current
+  # user object, or a Proc which will be passed the current user.
+  #
+  config.user.custom_data = {
+    :plan => Proc.new { |user| user.plan.slug },
+    :classrooms => Proc.new { |user| user.classrooms.count },
+    :memberships => Proc.new { |user| user.memberships.count },
```
Comment (stas) [inline config/initializers/intercom.rb]: Use `Coursewareable::User#memberships_count` to leverage from `:counter_cache => true`

File: config/initializers/intercom.rb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    :collaborations => Proc.new { |user| user.collaborations.count },
```
Comment (stas) [inline config/initializers/intercom.rb]: Same as above.

File: config/initializers/intercom.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  config.inbox.style = :default
+  # config.inbox.style = :custom
+
+  # == Inbox Counter
+  # If you're using the custom inbox style, you can request that Intercom
+  # insert an `em` element into your anchor with the count of unread messages
+  #
+  # config.inbox.counter = true
+
+end
```
Comment (stas) [inline config/initializers/intercom.rb]: `end if defined?(IntercomRails)` :+1:

File: Gemfile [hunk 6]
```diff
@@ -13,6 +13,7 @@ gem 'gravatar_image_tag'
 gem 'daemons'
 gem 'kaminari'
 gem 'ruby-oembed'
+gem 'intercom-rails'
```
Comment (stas) [inline Gemfile]: Move to production environment.

**Comment by dooma** [comment]
Done. I guess we should enable production environment, otherwise it won't work. But let put in production and we will see.

**Pull request merged**